### PR TITLE
regex: Plug a regex literal crash

### LINKED ISSUE #11937: Filtrer grep failed with pattern "/"
## Bug Report

**Describe the bug**
Filtrer grep failed with pattern "/"

If the pattern contains only one character and that character is "/", then fluent-bit crashes due to the parsing performed in the source code available here.
[Voir ici](https://github.com/fluent/fluent-bit/blob/master/src/flb_regex.c#L132C24-L132C25)

Furthermore, nowhere in the documentation does it specify that if the pattern begins with a "/" and ends with a "/", these two characters are removed.
This has an impact, for example, if you want to search for the text "/tmp/". In this case, you must search for "//tmp//".

### PR DESCRIPTION
**Pull request by cosmo0920**
Closes https://github.com/fluent/fluent-bit/issues/11937.

----
Enter `[N/A]` in the box, if an item is not applicable to your change.

**Testing**
Before we can approve your change; please submit the following in a comment:

- [ ] Example configuration file for the change
- [ ] Debug log output from testing the change

- [ ] Attached [Valgrind](https://valgrind.org/docs/manual/quick-start.html) output that shows no leaks or memory corruption was found

If this is a change to packaging of containers or native binaries then please confirm it works for all targets.

- [ ] Run [local packaging test](./packaging/local-build-all.sh) showing all targets (including any new ones) build.
- [ ] Set `ok-package-test` label to test for all targets (requires maintainer to do).

**Documentation**

- [ ] Documentation required for this feature

**Backporting**

- [ ] Backport to latest stable release.

----

Fluent Bit is licensed under Apache 2.0, by submitting this pull request I understand that this code will be released under the terms of that license.

## Bug Fixes
- Fixed an out-of-bounds access issue when processing short or empty regex patterns, improving stability and preventing potential crashes.

## Tests
- Added comprehensive test coverage for regex patterns with literal slashes to validate edge case handling.

### CODE REVIEW

**Comment by cosmo0920** [comment]
This Windows ARM64 fail https://github.com/fluent/fluent-bit/actions/runs/27324468341/job/80722539551?pr=11938 will be resolved with:
https://github.com/fluent/fluent-bit/pull/11934
For 4.2:
https://github.com/fluent/fluent-bit/pull/11935

File: tests/internal/regex.c [hunk 1]
```diff
@@ -319,12 +319,104 @@ static void test_option_i_x()
...
+        exit(1);
+    }
+}
+
+static void test_slash_delimited_literal_slashes()
+{
+    struct flb_regex *regex = NULL;
+    int ret;
+
+    regex = flb_regex_create("//tmp//");
```
Comment (patrick-stephens) [inline tests/internal/regex.c:384]: Hmm, so we have to use `//` to get one slash - should we have a different delimiter, e.g. `\/`?

Just worried we will end up with the nightmare of Java regexes and string escaping otherwise, e.g. if we want two slashes you have to have four, not three, etc.

Comment (cosmo0920) [reply tests/internal/regex.c:384]: No, we have to use double slash in the beginning and the end of regex literals. Not intending on double slash anywhere on the regex literals for our regex engine.

Comment (patrick-stephens) [reply tests/internal/regex.c:384]: Did we do a docs PR btw @cosmo0920 ?
